api/v1beta3/imagepolicy_types.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2023 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta3
+
+import (
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const ImagePolicyKind = "ImagePolicy"
+
+// Deprecated: Use ImageFinalizer.
+const ImagePolicyFinalizer = "finalizers.fluxcd.io"
+
+// ImagePolicySpec defines the parameters for calculating the
+// ImagePolicy.
+type ImagePolicySpec struct {
+	// ImageRepositoryRef points at the object specifying the image
+	// being scanned
+	// +required
+	ImageRepositoryRef meta.NamespacedObjectReference `json:"imageRepositoryRef"`
+	// Policy gives the particulars of the policy to be followed in
+	// selecting the most recent image
+	// +required
+	Policy ImagePolicyChoice `json:"policy"`
+	// FilterTags enables filtering for only a subset of tags based on a set of
+	// rules. If no rules are provided, all the tags from the repository will be
+	// ordered and compared.
+	// +optional
+	FilterTags *TagFilter `json:"filterTags,omitempty"`
+	// ReflectDigest governs the setting of the `.status.latestDigest` field.
```

Comment:
1. There is no semantic difference between `"Never"` and `nil` (see the test `TestImagePolicyReconciler_digestReflection`).
1. There are two main reasons for making this a pointer:
    1. If it weren't a pointer then any existing object's `.spec.digestReflectionPolicy` field would have an empty string as value when fetched from the API server by the controller. So either way we'd have to treat two values with the same semantics.
    2. Creating an `ImagePolicy` variable in Go without explicitly setting a value for `.spec.digestReflectionPolicy` would leave it empty (`""`). Hitting up the K8s API with that variable would fail if the empty string wouldn't be allowed. We'd have to change every single test to explicitly set an appropriate value here. It's much more convenient to allow `nil` and put proper semantics in place.